Getting EUNSUPPORTEDPROTOCOL error when creating project

### Version
4.5.3

### Environment info
```
Environment Info:

  System:
    OS: macOS 10.15.5
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 8.11.1 - ~/.nvm/versions/node/v8.11.1/bin/node
    Yarn: Not Found
    npm: 5.6.0 - ~/.nvm/versions/node/v8.11.1/bin/npm
  Browsers:
    Chrome: 84.0.4147.125
    Edge: Not Found
    Firefox: 79.0
    Safari: 13.1.1
  npmGlobalPackages:
    @vue/cli: 4.5.3
```


### Steps to reproduce
```
vue create my-project
select Default ([Vue 2] babel, eslint)
```

### What is expected?
It should create a new vue project

### What is actually happening?
I'm getting this error

```
Vue CLI v4.5.3
✨  Creating project in /Users/me/my-project.
🗃  Initializing git repository...
⚙️  Installing CLI plugins. This might take a while...

npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! Unsupported URL Type "npm:": npm:vue-loader@^16.0.0-beta.3

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2020-08-17T18_49_36_738Z-debug.log
 ERROR  command failed: npm install --loglevel error
```

---
This was working fine a few days ago, not sure what happened.

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/packages/@vue/cli/bin/vue.js
+++ b/packages/@vue/cli/bin/vue.js
@@ -19,12 +19,15 @@ function checkNodeVersion (wanted, id) {
 
 checkNodeVersion(requiredVersion, '@vue/cli')
 
-if (semver.satisfies(process.version, '9.x')) {
-  console.log(chalk.red(
-    `You are using Node ${process.version}.\n` +
-    `Node.js 9.x has already reached end-of-life and will not be supported in future major releases.\n` +
-    `It's strongly recommended to use an active LTS version instead.`
-  ))
+const EOL_NODE_MAJORS = ['8.x', '9.x']
+for (const major of EOL_NODE_MAJORS) {
+  if (semver.satisfies(process.version, major)) {
+    console.log(chalk.red(
+      `You are using Node ${process.version}.\n` +
+      `Node.js ${major} has already reached end-of-life and will not be supported in future major releases.\n` +
+      `It's strongly recommended to use an active LTS version instead.`
+    ))
+  }
 }
 
 const fs = require('fs')

--- a/packages/@vue/cli/bin/vue.js
+++ b/packages/@vue/cli/bin/vue.js
@@ -19,7 +19,7 @@ function checkNodeVersion (wanted, id) {
 
 checkNodeVersion(requiredVersion, '@vue/cli')
 
-const EOL_NODE_MAJORS = ['8.x', '9.x']
+const EOL_NODE_MAJORS = ['8.x', '9.x', '11.x', '13.x']
 for (const major of EOL_NODE_MAJORS) {
   if (semver.satisfies(process.version, major)) {
     console.log(chalk.red(

--- a/packages/@vue/cli/lib/Creator.js
+++ b/packages/@vue/cli/lib/Creator.js
@@ -28,6 +28,7 @@ const {
 const {
   chalk,
   execa,
+  semver,
 
   log,
   warn,
@@ -132,9 +133,10 @@ module.exports = class Creator extends EventEmitter {
       (hasYarn() ? 'yarn' : null) ||
       (hasPnpm3OrLater() ? 'pnpm' : 'npm')
     )
-    const pm = new PackageManager({ context, forcePackageManager: packageManager })
 
     await clearConsole()
+    const pm = new PackageManager({ context, forcePackageManager: packageManager })
+
     log(`✨  Creating project in ${chalk.yellow(context)}.`)
     this.emit('creation', { event: 'creating' })
 
@@ -172,6 +174,26 @@ module.exports = class Creator extends EventEmitter {
     await writeFileTree(context, {
       'package.json': JSON.stringify(pkg, null, 2)
     })
+
+    // generate a .npmrc file for pnpm, to persist the `shamefully-flatten` flag
+    if (packageManager === 'pnpm') {
+      const pnpmConfig = hasPnpmVersionOrLater('4.0.0')
+        ? 'shamefully-hoist=true\n'
+        : 'shamefully-flatten=true\n'
+
+      await writeFileTree(context, {
+        '.npmrc': pnpmConfig
+      })
+    }
+
+    if (packageManager === 'yarn' && semver.satisfies(process.version, '8.x')) {
+      // Vue CLI 4.x should support Node 8.x,
+      // but some dependenices already bumped `engines` field to Node 10
+      // and Yarn treats `engines` field too strictly
+      await writeFileTree(context, {
+        '.yarnrc': '# Hotfix for Node 8.x\n--install.ignore-engines true\n'
+      })
+    }
 
     // intilaize git repository before installing deps
     // so that vue-cli-service can setup git hooks.
@@ -232,17 +254,6 @@ module.exports = class Creator extends EventEmitter {
       log('📄  Generating README.md...')
       await writeFileTree(context, {
         'README.md': generateReadme(generator.pkg, packageManager)
-      })
-    }
-
-    // generate a .npmrc file for pnpm, to persist the `shamefully-flatten` flag
-    if (packageManager === 'pnpm') {
-      const pnpmConfig = hasPnpmVersionOrLater('4.0.0')
-        ? 'shamefully-hoist=true\n'
-        : 'shamefully-flatten=true\n'
-
-      await writeFileTree(context, {
-        '.npmrc': pnpmConfig
       })
     }
 

--- a/packages/@vue/cli/lib/util/ProjectPackageManager.js
+++ b/packages/@vue/cli/lib/util/ProjectPackageManager.js
@@ -109,7 +109,7 @@ class PackageManager {
     if (this.bin === 'npm') {
       // npm doesn't support package aliases until v6.9
       const MIN_SUPPORTED_NPM_VERSION = '6.9.0'
-      const npmVersion = execa.sync('npm', ['--version']).stdout
+      const npmVersion = stripAnsi(execa.sync('npm', ['--version']).stdout)
 
       if (semver.lt(npmVersion, MIN_SUPPORTED_NPM_VERSION)) {
         warn(

--- a/packages/@vue/cli/lib/util/ProjectPackageManager.js
+++ b/packages/@vue/cli/lib/util/ProjectPackageManager.js
@@ -104,6 +104,22 @@ class PackageManager {
       this.bin = loadOptions().packageManager || (hasYarn() ? 'yarn' : hasPnpm3OrLater() ? 'pnpm' : 'npm')
     }
 
+    if (this.bin === 'npm') {
+      // npm doesn't support package aliases until v6.9
+      const MIN_SUPPORTED_NPM_VERSION = '6.9.0'
+      const npmVersion = execa.sync('npm', ['--version']).stdout
+
+      if (semver.lt(npmVersion, MIN_SUPPORTED_NPM_VERSION)) {
+        warn(
+          'You are using an outdated version of NPM.\n' +
+          'there may be unexpected errors during installation.\n' +
+          'Please upgrade your NPM version.'
+        )
+
+        this.needsNpmInstallFix = true
+      }
+    }
+
     if (!SUPPORTED_PACKAGE_MANAGERS.includes(this.bin)) {
       log()
       warn(
@@ -308,6 +324,28 @@ class PackageManager {
         delete process.env.PUPPETEER_SKIP_CHROMIUM_DOWNLOAD
         await this.runCommand('install', ['--silent', '--no-progress'])
       }
+    }
+
+    if (this.needsNpmInstallFix) {
+      // if npm 5, split into several `npm add` calls
+      // see https://github.com/vuejs/vue-cli/issues/5800#issuecomment-675199729
+      const pkg = resolvePkg(this.context)
+      if (pkg.dependencies) {
+        const deps = Object.entries(pkg.dependencies).map(([dep, range]) => `${dep}@${range}`)
+        await this.runCommand('install', deps)
+      }
+
+      if (pkg.devDependencies) {
+        const devDeps = Object.entries(pkg.devDependencies).map(([dep, range]) => `${dep}@${range}`)
+        await this.runCommand('install', [...devDeps, '--save-dev'])
+      }
+
+      if (pkg.optionalDependencies) {
+        const devDeps = Object.entries(pkg.devDependencies).map(([dep, range]) => `${dep}@${range}`)
+        await this.runCommand('install', [...devDeps, '--save-optional'])
+      }
+
+      return
     }
 
     return await this.runCommand('install')

